### ISSUE TITLE
Manage deprecations...

### ISSUE THREAD

**Issue by kif** [inducer/pytools#27]
Hi Andreas,

All our CI start failing due to the drop of Python 3.5 in pytools.

Would it be possible to clearly state the deprecation of the library ?

Thanks,

Jérôme
```
Python 3.5.3 (default, Sep 27 2018, 17:25:39) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pytools
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jerome/workspace/pytools/build/lib/pytools/__init__.py", line 249
    __slots__: List[str] = []
             ^
SyntaxError: invalid syntax
```

**Comment by inducer** [comment]
Sorry about that! I think this duplicates #26, and it should be resolved as of last night (07aa9b5405cd23257497898dbcae05fae89e59fa). Please reopen/comment if that's not the case.

**Comment by kif** [comment]
Indeed, the bug is fixed. Sorry for the noise
